source/extensions/stat_sinks/common/hystrix/hystrix.cc

```diff
@@ -0,0 +1,320 @@
+  addIntToStream("currentMaximumPoolSize", 0, cluster_info);
+
+  ss << "data: {" << cluster_info.str() << "}" << std::endl << std::endl;
+}
+
+void Hystrix::getClusterStats(std::stringstream& ss, absl::string_view cluster_name,
+                              uint64_t max_concurrent_requests, uint64_t reporting_hosts,
+                              uint64_t rolling_window) {
+  addHystrixCommand(ss, cluster_name, max_concurrent_requests, reporting_hosts, rolling_window);
+  addHystrixThreadPool(ss, cluster_name, max_concurrent_requests, reporting_hosts, rolling_window);
+}
+
+absl::string_view Hystrix::printRollingWindow() const {
+  std::stringstream out_str;
+
+  for (auto stats_map_itr = rolling_stats_map_.begin(); stats_map_itr != rolling_stats_map_.end();
+       ++stats_map_itr) {
+    out_str << stats_map_itr->first << " | ";
+    RollingStats rolling_stats = stats_map_itr->second;
+    for (auto specific_stat_vec_itr = rolling_stats.begin();
+         specific_stat_vec_itr != rolling_stats.end(); ++specific_stat_vec_itr) {
+      out_str << *specific_stat_vec_itr << " | ";
+    }
+    out_str << std::endl;
+  }
+  return out_str.str();
+}
+
+namespace HystrixNameSpace {
+HystrixSink::HystrixSink(Server::Instance& server) : stats_(new Hystrix()), server_(&server){};
+
+void HystrixSink::beginFlush() { current_stat_values_.clear(); }
+
+void HystrixSink::flushCounter(const Stats::Counter& counter, uint64_t) {
+  if (callbacks_ == nullptr) {
+    return;
+  }
+  if (counter.name().find("upstream_rq_") != std::string::npos) {
+    current_stat_values_[counter.name()] = counter.value();
+  }
+}
+// void HystrixSink::flushGauge(const Gauge& gauge, uint64_t value);
+void HystrixSink::endFlush() {
+  if (callbacks_ == nullptr)
+    return;
+  stats_->incCounter();
+  for (auto& cluster : server_->clusterManager().clusters()) {
+    stats_->updateRollingWindowMap(current_stat_values_, cluster.second.get().info()->name());
+  }
+  std::stringstream ss;
+  for (auto& cluster : server_->clusterManager().clusters()) {
+    stats_->getClusterStats(
+        ss, cluster.second.get().info()->name(),
+        cluster.second.get()
+            .info()
+            ->resourceManager(Upstream::ResourcePriority::Default)
+            .pendingRequests()
+            .max(),
+        server_->stats()
+            .gauge("cluster." + cluster.second.get().info()->name() + ".membership_total")
+            .value(),
+        server_->statsFlushInterval().count());
+  }
+  Buffer::OwnedImpl data;
+  data.add(ss.str());
+  callbacks_->encodeData(data, false);
+
+  // send keep alive ping
+  Buffer::OwnedImpl ping_data;
+  ping_data.add(":\n\n");
+  callbacks_->encodeData(ping_data, false);
+}
+
+// void HystrixSink::onHistogramComplete(const Histogram& histogram, uint64_t value);
+void HystrixSink::registerConnection(Http::StreamDecoderFilterCallbacks* callbacks) {
+  callbacks_ = callbacks;
+}
+
+// TODO (@trabetti) is this correct way?
```

Comment:
I think to close the stream, you will need to make an `encodeData` (or some other encode* method) call on the callbacks with the `end_stream` argument set to `true`.

Reply:
we did not provide a way to terminate the stream from Envoy's side (should we?), this is called when the dashboard client was closed, so we are not sending any more data.